### PR TITLE
Moved init method into browserManager.

### PR DESCRIPTION
**Pull request by yl-coder**
Moved init method into browserManager. Created a start method to contain init tasks that require to be run in UI-thread.

### CODE REVIEW

**Comment by yl-coder** [comment]
@HansNewbie Can help to review this? thanks.

File: src/main/java/address/controller/MainController.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public MainController(MainApp mainApp, ModelManager modelManager, Config config)
     public void start(Stage primaryStage) {
         logger.info("Starting main controller.");
         this.primaryStage = primaryStage;
-        this.browserManager.initializeBrowser();
+        this.browserManager.start();
```
Comment (damithc) [inline src/main/java/address/controller/MainController.java]: Shouldn't this be `init()`?

**Comment by yl-coder** [comment]
There is some issue, hold on.

**Comment by HansNewbie** [comment]
I think you will need `BrowserManager.init()` where inside you will call `initJxBrowserEnv()`. Then call that `init` in `MainApp.init()`. I guess can keep the `start` to be consistent with `MainController.start`.

File: src/main/java/address/browser/BrowserManager.java [hunk 2]
```diff
@@ -58,7 +59,7 @@
     };
 
     public BrowserManager(ObservableList<ReadOnlyViewablePerson> filteredPersons) {
-        initializeBrowser();
+        initializeJxBrowserEnvironment();
```
Comment (damithc) [inline src/main/java/address/browser/BrowserManager.java]: Shouldn't this be conditional? i.e. only if the jxbrowser is being used

**Comment by damithc** [comment]
Added some prelim comments.

File: src/main/java/address/controller/MainController.java (hunk 1 above)
Comment (yl-coder) [inline src/main/java/address/controller/MainController.java]: I will have two method:
initBrowser() : to run init tasks in non-ui thread
start(): to run init tasks in ui thread

The start method in the MainController is in the ui thread.

File: src/main/java/address/browser/BrowserManager.java (hunk 2 above)
Comment (yl-coder) [inline src/main/java/address/browser/BrowserManager.java]: Calling BrowserManager.init() before the browserManager instance is quite odd. I'm calling the init after the browserManager instance is created at mainController.

Yes, I will add in the conditional.

File: src/main/java/address/browser/BrowserManager.java [hunk 3]
```diff
@@ -67,15 +67,18 @@ public BrowserManager(ObservableList<ReadOnlyViewablePerson> filteredPersons) {
      * Initialize the application to use jxBrowser.
      * This must be called in a non-ui thread.
      */
-    public static void initializeJxBrowserEnvironment(){
+    public static void initBrowser(){
```
Comment (HansNewbie) [inline src/main/java/address/browser/BrowserManager.java]: Update comments.

**Comment by HansNewbie** [comment]
Tell me if it is ready. I will test on Mac.

File: src/main/java/address/browser/BrowserManager.java [hunk 4]
```diff
@@ -57,25 +60,30 @@
...
     }
 
     /**
      * Initialize the application to use jxBrowser.
      * This must be called in a non-ui thread.
      */
-    public static void initializeJxBrowserEnvironment(){
-        if (Environment.isMac()) {
-            BrowserCore.initialize();
+    public void initBrowser(){
```
Comment (HansNewbie) [inline src/main/java/address/browser/BrowserManager.java]: Don't forget to update the header comment.

**Comment by yl-coder** [comment]
@HansNewbie Can help to try if the test case runs on ur mac

**Comment by HansNewbie** [comment]
Pass with flying colours.

**Comment by yl-coder** [comment]
@damithc Anymore comments before i merged this PR?

File: src/main/java/address/browser/BrowserManager.java [hunk 5]
```diff
@@ -58,32 +61,36 @@
...
-        if (Environment.isMac()) {
-            BrowserCore.initialize();
+    public static void initBrowser(){
+        if (type == BrowserType.FULL_FEATURE_BROWSER) {
+            if (Environment.isMac()) {
+                BrowserCore.initialize();
+            }
+            logger.debug("Suppressing browser logs");
+            LoggerProvider.setLevel(Level.SEVERE);
         }
```
Comment (damithc) [inline src/main/java/address/browser/BrowserManager.java]: This is better, but even better if you use a polymorphic call.
i.e. init() method is implemented in both types of browsers and you call the init() method here without knowing which type of a browser it is.

Comment (yl-coder) [inline src/main/java/address/browser/BrowserManager.java]: Do you mean to replace it with embeddedBrowser.init() here and implement the init method in respective concrete class of embeddedBrowser? The embeddedbrowser instance will only be created at the start() method. In which, the UI-thread will be started. And this block of codes need to run on a non-UI thread.

Comment (damithc) [inline src/main/java/address/browser/BrowserManager.java]: Yes, something like that. Never mind, we'll figure that out later.

**Comment by damithc** [comment]
No other comments from me.

**Pull request merged**